### ISSUE TITLE
Parameters for stored scripts not yet compatible with _update API of ES 6

### ISSUE THREAD

**Issue by thehgregor** [logstash-plugins/logstash-output-elasticsearch#638]
It seems that the `_update` API of ES 6.0.0.-beta1 does not accept parameter `lang` anymore:
```
POST _scripts/procdata-test
{
  "script": {
    "lang": "painless",
    "source": "ctx._source.xxx = params.event.get('process_id')"
  }
}
```
```
POST staging-procdata/procdata-order/42/_update
{
  "script": {
    "id": "procdata-test",
    "lang": "painless"
  }
}
```
results in:
`HTTP 400 - illegal_argument_exception: illegally specified <lang> for a stored script`

Parameter `lang` is not accepted anymore and needs to be omitted.

Logstash's elasticsearch-output-plugin does not adhere to this yet, and any attempt to use a scripted update results in the same error as above. Parameter `lang` may not be passed to the ES `_update` API anymore in the case of stored scripts (`script_type => indexed`).

(Note: setting `script_lang` to `null` explicitly in the Logstash configuration does not work, so this must be fixed in the code of the output plugin)

- Version: 6.0.0-beta1
- Operating System: CentOS 7
- Config File (if you have sensitive info, please remove it):
```
output {
  elasticsearch {
    "action" => "update"
    "hosts" => ["elasticsearch:9200"]
    "index" => "staging-procdata"
    "document_id" => "%{process_id}"
    "doc_as_upsert" => true
    "script_var_name" => "event"
    "script" => "procdata-test"
    "script_type" => "indexed"
  }
}
```
- Sample Data: any
- Steps to Reproduce: Create a script in ES and try to use it in a scripted update of a document with above configuration of the logstash-output-elasticsearch plugin.

**Comment by jordansissel** [comment]
Thank you for reporting. We will investigate as time permits.

**Comment by untergeek** [comment]
Hmm. No pioneer tag in this repo, but I think this is worthy

**Comment by andrewvc** [comment]
I'm thinking that the solution here is going to involve actual version sniffing on the live cluster. We don't want to be requiring people to explicitly declare a null option, or declare their ES version etc.

**Comment by jsvd** [comment]
we already [fetch the version of the node we hit first when in the template manager](https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/e7259de52f88643cc3311ce73d4dfa4663041a87/lib/logstash/outputs/elasticsearch/template_manager.rb#L15), this should probably be pulled out of that class

**Comment by andrewvc** [comment]
This looks like it will be fixed with: https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/642

**Comment by rfraposa** [comment]
I noticed #642 does not have a milestone yet. Was this issue fixed for 6.00-rc1?

**Comment by andrewvc** [comment]
@rfraposa this won't make RC1 but should make RC2
